soc/arm/atmel_sam0/common/soc_samd2x.c

```diff
@@ -246,6 +246,20 @@ static inline void gclk_adc_configure(void)
 }
 #endif
 
+#if !CONFIG_WDT_SAM0
+#define gclk_wdt_configure()
+#else
+static inline void gclk_wdt_configure(void)
+{
+	GCLK->GENDIV.reg = GCLK_GENDIV_ID(2) | GCLK_GENDIV_DIV(4);
+	GCLK->GENCTRL.reg = GCLK_GENCTRL_ID(2) | GCLK_GENCTRL_GENEN |
+						GCLK_GENCTRL_SRC_OSCULP32K | GCLK_GENCTRL_DIVSEL;
+
+	while (GCLK->STATUS.bit.SYNCBUSY) {
+	}
+}
+#endif
+
```

Comment:
I think positive logic is easy to read. BTW, you need align correct as below:

```c
#if CONFIG_WDT_SAM0
static inline void gclk_wdt_configure(void)
{
	GCLK->GENDIV.reg = GCLK_GENDIV_ID(2)
			 | GCLK_GENDIV_DIV(4);

	GCLK->GENCTRL.reg = GCLK_GENCTRL_ID(2)
			  | GCLK_GENCTRL_GENEN
			  | GCLK_GENCTRL_SRC_OSCULP32K
			  | GCLK_GENCTRL_DIVSEL;

	while (GCLK->STATUS.bit.SYNCBUSY) {
	}
}
#else
#define gclk_wdt_configure()
#endif
```

I was wondering if below is more readable

```c
static inline void gclk_wdt_configure(void)
{
	if (IS_ENABLED(CONFIG_WDT_SAM0)) {
		GCLK->GENDIV.reg = GCLK_GENDIV_ID(2)
				 | GCLK_GENDIV_DIV(4);

		GCLK->GENCTRL.reg = GCLK_GENCTRL_ID(2)
				  | GCLK_GENCTRL_GENEN
				  | GCLK_GENCTRL_SRC_OSCULP32K
				  | GCLK_GENCTRL_DIVSEL;

		while (GCLK->STATUS.bit.SYNCBUSY) {
		}
	}
}
```